cypress/integration/post_header/post_header_spec.js

```diff
@@ -158,4 +159,58 @@ describe('Post Header', () => {
         // * Check that the RHS is close
         cy.get('#rhsContainer').should('not.be.visible');
     });
+
+    it('should add then remove pinned badge in both center channel and RHS flagged posts list', () => {
+        // 2. Go to a known team and channel
+        cy.visit('/ad-1/channels/town-square');
+
+        // 3. Post a message
+        cy.postMessage('test for pinning/unpinning a post{enter}');
+
+        cy.getLastPostId().then((postId) => {
+            // * Check that the center flag icon of the post is not visible
+            cy.get(`#CENTER_flagIcon_${postId}`).should('not.be.visible');
+
+            // * Check that the pinned badge of the post is not visible
+            cy.get(`#post_${postId}`).should('not.have.class', 'post--pinned');
+
+            // 4. Pin the post.
+            cy.clickPostDotMenu(postId);
+            cy.get(`#pin_post_${postId}`).should('be.visible').click();
+
+            // 5. Click the center flag icon of a post
+            cy.clickPostFlagIcon(postId);
+
+            // 6. click RHS list
+            cy.get('#channelHeaderFlagButton').click({force: true});
+
+            // * Check that message exists in RHS flagged posts list
+            cy.get(`#postMessageText_${postId}`).contains('test for pinning/unpinning a post');
+
+            // * Check that post is be pinned in center
+            // * Check that post is be pinned in RHS
+            cy.get(`#post_${postId}`).should('have.class', 'post--pinned');
+            cy.get(`#SEARCH_flagIcon_${postId}`).siblings('.post__pinned-badge').should('exist');
+
+            // 7. unpin the post
+            cy.clickPostDotMenu(postId);
+            cy.get(`#unpin_post_${postId}`).should('be.visible').click();
+
+            // * Check that message exists in RHS flagged posts list
+            // * Check that post is not be pinned in center
+            // * Check that post is not be pinned in RHS
+            cy.get(`#postMessageText_${postId}`).contains('test for pinning/unpinning a post');
+            cy.get(`#post_${postId}`).should('not.have.class', 'post--pinned');
+            cy.get(`#SEARCH_flagIcon_${postId}`).siblings('.post__pinned-badge').should('not.exist');
+
+            // 8. Pin the post again.
+            cy.clickPostDotMenu(postId);
+            cy.get(`#pin_post_${postId} button`).should('be.visible').click({force: true});
+
+            // * Check that post is be pinned in center
+            // * Check that post is be pinned in RHS
+            cy.get(`#post_${postId}`).should('have.class', 'post--pinned');
+            cy.get(`#SEARCH_flagIcon_${postId}`).siblings('.post__pinned-badge').should('exist');
```

Comment:
Maybe is not important, would be great if we leave the state of the application as similar as possible to the initial state. Should we remove the created post here? @saturninoabril you have more experience here, is something that is already done automatically or something like that?

Reply:
We're doing that in user and system settings but not on post (like deleting a post after testing). While it's ideal, I'm on a fence of actually doing it since it will prolong test duration. Note that when we are doing release testing, we don't have step to delete that as well unless specifically required.